nav-challenge/src/store/user/UserReducer.ts

```diff
@@ -0,0 +1,37 @@
+import { UserDataType } from "../../utils/types/UserDataType";
+import { SIGNUP_USER } from "./UserActionTypes";
+
+const initialState: UserDataType = {
+  currentUser: {
+    token: "",
+    username: "",
+    first_name: "SASA",
+    last_name: "",
+    user_id: 0,
+    email: "",
+    isInfluencer: false,
+    hasSelectedInfluencers: false,
+    country: "",
+    city: "",
+    bio: "",
+    jobTitle: "",
+    numberOfFollowers: "",
+    numberOfFollowings: "",
+    profileImage: "",
+    coverImage: "",
+  },
+};
+
+const user = (state = initialState, action: any = {}): UserDataType => {
+  switch (action.type) {
+    case SIGNUP_USER:
+      return {
```

Comment:
Shouldn't we be doing something with the `action` as well?

Reply:
not sure if i understand...

Reply:
Since we have 2 arguments to the `userReducer` function (`state`, `action`) we are only returning `{...state}`, but do we need to do something with the `action` parameter as well?

Reply:
in this case action is being used in switch statement `switch (action.type)` to check what type of action we should process.